app/gui/src/project-view/util/ast/abstract.ts

```diff
@@ -288,9 +288,12 @@ export function dropMutability<T extends Ast>(value: Owned<Mutable<T>>): T {
   return value as unknown as T
 }
 
-function unwrapGroups(ast: Ast) {
-  while (ast instanceof Group && ast.expression) ast = ast.expression
-  return ast
+/**
+ * If the input is a parenthesized expression, returns the inner expression; otherwise, returns the
+ * input.
+ */
+export function unwrapGroups<T extends Ast | undefined>(ast: T): T | Expression {
+  return ast instanceof Group && ast.expression ? unwrapGroups(ast.expression) : ast
```

Comment:
Loop implementation might be a bit better for js engine... unless it is optimize by esbuild or something.

Reply:
Because this will almost never need to recurse more than once (outside of pathological inputs), IMO simplicity is worth more than saving stack frames here.